app/src/main/kotlin/github/com/rhacco/dota2androidapp/activities/LiveMatchesActivity.kt

```diff
@@ -0,0 +1,101 @@
+package github.com.rhacco.dota2androidapp.activities
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import github.com.rhacco.dota2androidapp.R
+import github.com.rhacco.dota2androidapp.lists.LiveMatchesListAdapter
+import github.com.rhacco.dota2androidapp.sources.remote.getDota2OfficialAPIService
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_live_matches.*
+
+class LiveMatchesActivity : AppCompatActivity() {
+    private lateinit var mListAdapter: LiveMatchesListAdapter
+    private val mDisposables = CompositeDisposable()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_live_matches)
+        actionBar?.setDisplayHomeAsUpEnabled(true)
+        mListAdapter = LiveMatchesListAdapter(this)
+        listView.adapter = mListAdapter
+        updateLiveMatches()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mDisposables.clear()
+    }
+
+    private fun updateLiveMatches() {
+        val disposable = getDota2OfficialAPIService()
```

Comment:
Why are we directly exposing the reactive strucutre? Why aren't we using a viewmodel for this?

Showing the reactive structure completely defeats the purpose of the viewmodel architecture

Reply:
Fixed